### PR TITLE
ci/openshift-ci: Port the job to run on OCP 4.14

### PR DESCRIPTION
**Pull request by ldoktor**
The OCP 4.14 uses pod security admission to disallow running privileged containers by default. This commit sets the right labels to allow running them and modifies the pattern to use the same stock qemu with ocp versions "4.*" as it should be compatible across multiple releases.

Fixes: https://github.com/kata-containers/tests/issues/5671

@wainersm does this sound right?

### CODE REVIEW

**Comment by ldoktor** [comment]
Tested locally on ocp 4.14:

```
$ bash .ci/openshift-ci/test.sh
Client Version: 4.8.0-202111041632.p0.git.88e7eba.assembly.stream-88e7eba
Server Version: 4.14.0-0.nightly-2023-05-23-103225
Kubernetes Version: v1.27.1+38c64ac
INFO: Install and configure kata into the test cluster
Now using project "default" on server "XXX".
~/go/src/github.com/kata-containers/kata-containers ~/go/src/github.com/kata-containers/tests
INFO: Applying kata-deploy
serviceaccount/kata-label-node created
clusterrole.rbac.authorization.k8s.io/node-labeler created
clusterrolebinding.rbac.authorization.k8s.io/kata-label-node-rb created
namespace/kube-system labeled
W0526 10:13:25.940736 1131576 warnings.go:70] would violate PodSecurity "baseline:latest": hostPath volumes (volumes "crio-conf", "containerd-conf", "kata-artifacts", "dbus", "systemd", "local-bin"), privileged (container "kube-kata" must not set securityContext.privileged=true)
daemonset.apps/kata-deploy created
pod/kata-deploy-679l5 condition met
pod/kata-deploy-dft8w condition met
pod/kata-deploy-dpw8h condition met
INFO: Adding the kata runtime classes
runtimeclass.node.k8s.io/kata-qemu created
runtimeclass.node.k8s.io/kata-qemu-sev created
runtimeclass.node.k8s.io/kata-qemu-tdx created
runtimeclass.node.k8s.io/kata-qemu-snp created
runtimeclass.node.k8s.io/kata-clh created
runtimeclass.node.k8s.io/kata-fc created
runtimeclass.node.k8s.io/kata-dragonball created
runtimeclass.node.k8s.io/kata-qemu-nvidia-gpu created
~/go/src/github.com/kata-containers/tests
INFO: Configuring SELinux
machineconfig.machineconfiguration.openshift.io/51-kata-selinux created
NAME              GENERATEDBYCONTROLLER   IGNITIONVERSION   AGE
51-kata-selinux                           2.2.0             1s
Wait ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus21-6t85l reboot
Wait ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus22-qblnj reboot
Wait ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus23-7ktmz reboot
Set timeout to 900 seconds
Checking after 60 seconds
Checking after 122 seconds
Checking after 184 seconds
ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus21-6t85l rebooted
Checking after 245 seconds
Checking after 307 seconds
ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus22-qblnj rebooted
Checking after 368 seconds
Checking after 429 seconds
Checking after 489 seconds
Checking after 550 seconds
ci-ln-9rvwr0t-1d09d-6d5wj-worker-eastus23-7ktmz rebooted
INFO: Run test suite: smoke
INFO: Creating the http-server pod
pod/http-server created
INFO: Wait until the pod gets running
INFO: http-server is running
INFO: Creating the service and route
service/http-server-service created
route.route.openshift.io/http-server-route created
INFO: HTTP server is working
INFO: Deleting resources created
service "http-server-service" deleted
route.route.openshift.io "http-server-route" deleted
INFO: Deleting the http-server pod
pod "http-server" deleted
INFO: Test suite: smoke: PASS
```

**Comment by wainersm** [comment]
> The OCP 4.14 uses pod security admission to disallow running privileged containers by default. This commit sets the right labels to allow running them and modifies the pattern to use the same stock qemu with ocp versions "4.*" as it should be compatible across multiple releases.
>

Hey, @ldoktor , thanks for your first contribution to Kata Containers!
 
> @wainersm does this sound right?

It does!

I ran the static checks on this PR and there is one thing to change: at least one commit of the series must have a `Fixes #NUMBER` referencing an issue number.  I checked here and it seems I didn't open an issue for the problem you solved, so could you please open it yourself?

The https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md has more details.

Ah, devels of kata containers usually force push  to overwrite the old PR in case of new changes, so don't need to open a new v2, v3...vN PRs. However, it is up to you...if you prefer to open a PR for each change, do it.

**Review by wainersm — APPROVED** [review]
LGTM. Thanks @ldoktor !

File: .ci/openshift-ci/cluster/install_kata.sh [hunk 1]
```diff
@@ -166,7 +167,7 @@ if [ "${KATA_WITH_SYSTEM_QEMU}" == "yes" ]; then
 		4.10*)
 			oc apply -f ${deployments_dir}/configmap_installer_qemu_4_10.yaml
 			;;
-		4.11*)
+		4.*)
```
[the remark this answers is not in the document]
Comment (wainersm) [reply .ci/openshift-ci/cluster/install_kata.sh:170]: You are correct on your rationale, @gkurz . The case for older OCP versions can be removed.

**Comment by ldoktor** [comment]
* kubectl -> oc
* removed 4.10 custom config and moved to a separate commit

**Review by gkurz — APPROVED** [review]
LGTM. Thanks @ldoktor !

**Comment by ldoktor** [comment]
> LGTM. Thanks @ldoktor !

Thank you, Greg, is there anything left to get this merged? I have a related fix to https://github.com/openshift/release to start testing with 4.14 that would be nice to get in... (note @wainersm is on PTO, should I ping someone else to get a 2nd review of the updated version?)

**Comment by ldoktor** [comment]
@wainersm no code changes, just rebase

**Comment by wainersm** [comment]
@ldoktor I am checking with the community what is going on with those ARM and s390x jobs. They should not be failing as you changes don't touch them.

**Comment by jongwu** [comment]
The error on Arm is nothing with here. I reaise a [PR](https://github.com/kata-containers/kata-containers/pull/7188) to fix it.

**Comment by wainersm** [comment]
@ldoktor a fix has been merged (https://github.com/kata-containers/kata-containers/pull/7188). Could you please rebase this PR?

**Pull request merged**